### DISCUSSION TITLE
Create branch with —no-track option

### DISCUSSION THREAD

**Discussion by sflomenb** [magit/magit#4358] [Support] [answered]
Hi. I thought I’d reach out here instead/before creating an issue.

Let me start off by saying how amazing magit is. It is the reason I started using emacs in the first place. As a long-time user of the git CLI, I was opposed to using anything else. But magit makes things easier and faster.

My typical workflow on the CLI is I would create a new branch from master. However, instead of master, it would usually be origin/master since the local branch might not have all the changes the remote ones does.

I would do this like so to create `feature/foo` branch:

`git checkout -b feature/foo origin/master —no-track`

The important part here is `—no-track`, as I don’t want the upstream to be origin/master. I want it to be unset and I can create it later when I am ready to push my branch.

How can I do this with magit? What I’ve been doing is 

`b c`, type in `feature/foo` and then once the branch is created, I would do `b u` to unset the remote from `origin/master` on the new branch.

I’ve read the manual, the FAQs, etc. What I am looking for is the ability to set this option ad-hoc. Similar to when you commit or use the log in magit, you can set options just for that once use of the command.

Let me know your thoughts, if this is possible, and if I should potentially create an issue for this.

Thank you.

**Comment by tarsius** [comment, marked as answer]
> How can I do this with magit?

You can manually append `^0` to the input, e.g. `origin/master^0`. This will dereference the branch as a commit, preventing it from being used as the upstream branch.

But this is one of those "you don't actually want that" situations. Please read https://magit.vc/manual/magit/The-Two-Remotes.html.
